### PR TITLE
Add ECS Service of Fargate launch type with Application Auto Scaling …

### PR DESCRIPTION
**Pull request by uttarasridhar**
Example for a Fargate Service with Application Auto Scaling using target tracking policies.

Follows the [AWS Docs example](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-autoscaling-targettracking.html#targettracking-tutorial) for setting up the service, scaling policy and performing a load test to trigger scaling activity.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: typescript/ecs/fargate-service-with-auto-scaling/index.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import appautoscaling = require('@aws-cdk/aws-applicationautoscaling');
+import ecs = require('@aws-cdk/aws-ecs');
+import ec2 = require('@aws-cdk/aws-ec2');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-fargate-application-autoscaling');
```
Comment (eladb) [inline typescript/ecs/fargate-service-with-auto-scaling/index.ts]: Can we please rewrite it more idiomatically by defining a class that extends `cdk.Stack` and always use `this` as the `scope` argument.

Comment (uttarasridhar) [reply typescript/ecs/fargate-service-with-auto-scaling/index.ts]: done.

**Comment by PaulMaddox** [comment]
I think this example could be simplified further:

```javascript
const scaling = service.autoScaleTaskCount({ maxCapacity: 10 });
scaling.scaleOnCpuUtilization('CpuScaling', {
  targetUtilizationPercent: 50
});
```

Source: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/README.md#task-auto-scaling

I know when I started looking at this, it was not obvious to me that the service.autoScaleTaskCount() method returned an object that could help me with setting a scaling policy. 

Maybe that’s something we can improve.

**Comment by uttarasridhar** [comment]
Thanks for the pointer @PaulMaddox. This exemplifies how these higher level cdk constructs simplify the common workflows.

**Pull request merged**